Mojave build

milljm opened the pull request.

Commit:
format the build version

format the build version so it is easier to understand within civet.

Commit:
Changes necessary to for a osx mojave build

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/237132/) on fee6b07 wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/164/site)

This comment will be updated on new commits.

Commit:
Wait for LLVM to be built

Commit:
Build LLVM correctly (borrowed from another PR)

Modify GCC config to include jit and several other switches 'borrowed' from the homebrew
recipes. I trust their configure switches more than what I am finding on my own.

Commit:
Create new jit-patch for darwin targets

Commit:
Mojave is special..

Commit:
Fix syntax typo

Perform jit darwin patch as a function

Commit:
Okay, now Mojave is special...

Commit:
Something wierd with the patch file

I created the initial patch file on Mojave. For this attempt, I used High Sierra.
Interesting diff on this PR (minus the directory structure, which I did change).

It may be that `diff -u` created a different patch while on Mojave than it did for High Sierra?

milljm commented:
Merging due to just one app with 4 tests failing...
